packages/components/pie-icon-button/test/accessibility/pie-icon-button.spec.ts

```diff
@@ -5,26 +5,45 @@ import type {
 import {
     getAllPropCombinations, splitCombinationsByPropertyValue,
 } from '@justeattakeaway/pie-webc-testing/src/helpers/get-all-prop-combos.ts';
+import { IconClose } from '@justeattakeaway/pie-icons-webc';
 import { PieIconButton } from '../../src/index.ts';
 import { variants } from '../../src/defs.ts';
 
 const props: PropObject = {
     variant: variants,
     disabled: [true, false],
+    aria: {
+        labelledby: 'foo',
+        describedby: 'bar',
+        controls: 'baz',
+        expanded: false,
+        label: 'qux',
+    },
 };
 
 const componentPropsMatrix: WebComponentPropValues[] = getAllPropCombinations(props);
 const componentPropsMatrixByVariant: Record<string, WebComponentPropValues[]> = splitCombinationsByPropertyValue(componentPropsMatrix, 'variant');
 const componentVariants: string[] = Object.keys(componentPropsMatrixByVariant);
 
+test.beforeEach(async ({ mount }, testInfo) => {
+    testInfo.setTimeout(testInfo.timeout + 40000);
+
+    // This ensures the icon-button and icon components are registered in the DOM for each test.
+    // It appears to add them to a Playwright cache which we understand is required for the tests to work correctly.
+    const iconButtonComponent = await mount(PieIconButton);
+    await iconButtonComponent.unmount();
+    const iconComponent = await mount(IconClose);
+    await iconComponent.unmount();
```

Comment:
I assume there are other occurrences of this workaround in other test related files.

It would be nice to have a shared function that abstracts this workaround with something less verbose, that could be easily removed if and when it becomes unnecessary.

Something along these lines...
```
async function mountComponentsForPlaywright(componentsClassArray) {
  for await (const component of componentsClassArray) {
    const mountedComponent = await mount(component);
    await mountedComponent.unmount();
  }
}
```

Reply:
I agree and think that would be a nice change. Would you be happy for me to raise a separate PR after this that applies that change to all existing test files @fernandofranca ?

Reply:
That would be great! 👍